client/components/time-since/index.jsx

```diff
@@ -2,6 +2,16 @@ import { useMemo } from 'react';
 import { useLocalizedMoment } from 'calypso/components/localized-moment';
 import { useHumanDate } from 'calypso/lib/human-date';
 
+/**
+ * @typedef {Object}   TimeSinceProps
+ * @property {string}  date       The timestamp to render
+ * @property {string=} dateFormat Moment format string for the tooltip, default = 'll'
+ * @property {string=} className  Forwarded to underlying <time> element
+ */
+
+/**
+ * @param {TimeSinceProps} props
+ */
```

Comment:
As we're adding types to this function, doesn't it make sense to refactor it to be a TSX file?

Reply:
@zaguiini I refactored it to TSX. I'm not sure which way of handling the default prop value is preferred, so I kept the one closer to the standard JS way.

Reply:
> I'm not sure which way of handling the default prop value is preferred

Yup the standard JS way of doing default props works in React. It's one of the things I prefer about functional components over the older `MyComp extends React.Component` way of writing components.